tests/View/ViewFactoryTest.php

```diff
@@ -427,6 +427,101 @@ public function testExtraAppendSectionCallThrowsException()
         $factory->appendSection();
     }
 
+    public function testAddingLoops()
+    {
+        $factory = $this->getFactory();
+
+        $factory->addLoop([1, 2, 3]);
+
+        $expectedLoop = [
+            'index' => 0,
+            'reverseIndex' => 4,
+            'size' => 3,
+            'isFirst' => true,
+            'isLast' => false,
+            'depth' => 1,
+            'parent' => null,
+        ];
+
+        $this->assertEquals([$expectedLoop], $factory->getLoopsStack());
+
+        $factory->addLoop([1, 2, 3, 4]);
+
+        $secondExpectedLoop = [
+            'index' => 0,
+            'reverseIndex' => 5,
+            'size' => 4,
+            'isFirst' => true,
+            'isLast' => false,
+            'depth' => 2,
+            'parent' => $expectedLoop,
+        ];
+
+        $this->assertEquals([$expectedLoop, $secondExpectedLoop], $factory->getLoopsStack());
+    }
+
+    public function testAddingUncountableLoop()
+    {
+        $factory = $this->getFactory();
+
+        $factory->addLoop('');
+
+        $expectedLoop = [
+            'index' => 0,
+            'reverseIndex' => null,
+            'size' => null,
+            'isFirst' => true,
+            'isLast' => null,
+            'depth' => 1,
+            'parent' => null,
+        ];
+
+        $this->assertEquals([$expectedLoop], $factory->getLoopsStack());
+    }
+
+    public function testIncrementingLoopIndices()
+    {
+        $factory = $this->getFactory();
+
+        $factory->addLoop([1, 2, 3, 4]);
+
+        $factory->incrementLoopIndices();
+
+        $factory->incrementLoopIndices();
+
+        $this->assertEquals(2, $factory->getLoopsStack()[0]['index']);
+        $this->assertEquals(3, $factory->getLoopsStack()[0]['reverseIndex']);
+    }
+
```

Comment:
I got an idea that you can test `$loop` in view by assert `$factory->shared('loop')`, for example
```
$this->assertEquals(2, $factory->shared('loop')->index)
```